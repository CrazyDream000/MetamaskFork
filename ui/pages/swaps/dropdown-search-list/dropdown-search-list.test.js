import React from 'react';
import configureMockStore from 'redux-mock-store';

import {
  renderWithProvider,
  createSwapsMockStore,
} from '../../../../test/jest';
import DropdownSearchList from '.';

const createProps = (customProps = {}) => {
  return {
    startingItem: {
      iconUrl: 'iconUrl',
      symbol: 'symbol',
    },
    ...customProps,
  };
};

describe('DropdownSearchList', () => {
  it('renders the component with initial props', () => {
    const store = configureMockStore()(createSwapsMockStore());
    const props = createProps();
    const { container, getByText } = renderWithProvider(
      <DropdownSearchList {...props} />,
      store,
    );
    expect(container).toMatchSnapshot();
    expect(getByText('symbol')).toBeInTheDocument();
  });
});
