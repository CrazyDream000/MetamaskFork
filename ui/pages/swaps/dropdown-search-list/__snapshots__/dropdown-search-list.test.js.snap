// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownSearchList renders the component with initial props 1`] = `
<div>
  <div
    class="dropdown-search-list"
    tabindex="0"
  >
    <div
      class="dropdown-search-list__selector-closed-container"
    >
      <div
        class="dropdown-search-list__selector-closed"
      >
        <img
          alt=""
          class="url-icon dropdown-search-list__selector-closed-icon"
          src="iconUrl"
        />
        <div
          class="dropdown-search-list__labels"
        >
          <div
            class="dropdown-search-list__item-labels"
          >
            <span
              class="dropdown-search-list__closed-primary-label"
            >
              symbol
            </span>
          </div>
        </div>
      </div>
      <i
        class="fa fa-caret-down fa-lg dropdown-search-list__caret"
      />
    </div>
  </div>
</div>
`;
